lib/sparse/src/common/sparse_vector.rs

```diff
@@ -85,7 +85,55 @@ impl SparseVector {
             None
         }
     }
+
+    /// Construct a new vector that is the result of performing all indices-wise operations
+    pub fn combine_aggregate(
+        &self,
+        other: &SparseVector,
+        op: impl Fn(DimWeight, DimWeight) -> DimWeight,
+    ) -> Self {
+        debug_assert!(self.is_sorted());
+        debug_assert!(other.is_sorted());
+
+        let mut result = SparseVector::default();
+        let mut i = 0;
+        let mut j = 0;
+        while i < self.indices.len() && j < other.indices.len() {
+            match self.indices[i].cmp(&other.indices[j]) {
+                std::cmp::Ordering::Less => {
+                    result.indices.push(self.indices[i]);
+                    result.values.push(op(self.values[i], 0.0));
+                    i += 1;
+                }
+                std::cmp::Ordering::Greater => {
+                    result.indices.push(other.indices[j]);
+                    result.values.push(op(other.values[j], 0.0));
```

Comment:
`op` agruments are not simmetric. It should be `op(0.0, other.values[j])`

Reply:
fixed. also added test for this case